storage/mysql/admin_storage.go

```diff
@@ -161,7 +158,7 @@ func (t *adminTX) GetTree(ctx context.Context, treeID int64) (*trillian.Tree, er
 	defer stmt.Close()
 
 	// GetTree is an entry point for most RPCs, let's provide somewhat nicer error messages.
-	tree, err := readTree(stmt.QueryRowContext(ctx, treeID))
+	tree, err := storage.ReadTree(stmt.QueryRowContext(ctx, treeID))
```

Comment:
I don't know if this is a good idea or not...
OTOH it does promote reuse and reduce boilerplate, but on the other it forces some restrictions on implementations, maybe that's a good trade off to make until we discover a scenario where it doesn't work, let's see what others think.

@pphaneuf @Martin2112 @RJPercival  @daviddrysdale

Reply:
Yeah, definitely wasn't sure how much to pull out/leave in. My general understanding was that `db/sql` abstracted _most_ differences away to only being at the query level but I could be wrong about that (especially with types, etc.)

Reply:
Yeah, definitely wasn't sure how much to pull out/leave in. My general understanding was that db/sql abstracted most differences away to only being at the query level but I could be wrong about that (especially with types, etc.)